src/SessionSetup/ConnectToStadiaWidget.cpp

```diff
@@ -561,31 +423,45 @@ void ConnectToStadiaWidget::TrySelectRememberedInstance() {
   remembered_instance_id_ = std::nullopt;
 }
 
-void ConnectToStadiaWidget::showEvent(QShowEvent* event) {
-  QWidget::showEvent(event);
-  // It is important that the call to DetachRadioButton is done here and not during construction.
-  // For high dpi display settings in Windows (scaling) the the actual width and height of the radio
-  // button is not known during construction. Hence the call is done when the widget is shown, not
-  // when its constructed
-  DetachRadioButton();
-}
-
 bool ConnectToStadiaWidget::IsActive() const { return ui_->contentFrame->isEnabled(); }
 
-void ConnectToStadiaWidget::SetProject(const std::optional<orbit_ggp::Project>& project) {
-  if (selected_project_ == project) return;
+void ConnectToStadiaWidget::Connect() {
+  CHECK(selected_instance_.has_value());
 
-  QSettings settings;
-  if (!project.has_value()) {
-    settings.setValue(kSelectedProjectDisplayNameKey, QVariant());
-    settings.setValue(kSelectedProjectIdKey, QVariant());
-  } else {
-    settings.setValue(kSelectedProjectDisplayNameKey, project.value().display_name);
-    settings.setValue(kSelectedProjectIdKey, project.value().id);
+  auto account_result = GetAccountSync();
+  if (account_result.has_error()) {
+    emit ErrorOccurred(QString::fromStdString(account_result.error().message()));
+    return;
+  }
+
+  if (!absl::StartsWith(account_result.value().email.toStdString(),
+                        selected_instance_->owner.toStdString())) {
+    OtherUserDialog dialog{selected_instance_->owner, this};
+    auto dialog_result = dialog.Exec();
+    if (dialog_result.has_error()) return;
   }
 
-  selected_project_ = project;
-  emit InstanceReloadRequested();
+  emit Connecting();
+}
+
+ErrorMessageOr<Account> ConnectToStadiaWidget::GetAccountSync() {
+  if (cached_account_ != std::nullopt) return cached_account_.value();
+
+  // This async call is not doing network calls, only reads local config files, hence it is fast and
+  // used in a syncronous manner here.
+  QEventLoop local_loop;
+  auto account_future = ggp_client_->GetDefaultAccountAsync().Then(
+      main_thread_executor_.get(), [&local_loop](ErrorMessageOr<Account> account) {
+        local_loop.quit();
+        return account;
+      });
+  local_loop.exec();
+
+  OUTCOME_TRY(auto&& account, account_future.Get());
+
+  cached_account_ = std::move(account);
```

Comment:
You can combine those: `OUTCOME_TRY(cached_account_, account_future.Get());`

Reply:
thanks. Done